app/models/respondent.rb

```diff
@@ -11,17 +11,11 @@ class Respondent < ActiveRecord::Base
   before_save :enqueue_fee_group_reference_request, if: :needs_fee_group_reference?
 
   def address
-    addresses.first
+    addresses.detect(&:primary?) || addresses.build(primary: true)
```

Comment:
Would have preferred to use `find_or_initialize_by` but these query methods don't consider unpersisted members of the collection, i.e. they just make a DB query.